gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/ManifestBasedDataset.java

```diff
@@ -128,6 +143,12 @@ public Iterator<FileSet<CopyEntity>> getFileSetIterator(FileSystem targetFs, Cop
           toDelete.add(targetFs.getFileStatus(fileToCopy));
         }
       }
+
+      Properties props = new Properties();
+      props.setProperty(SetPermissionCommitStep.STOP_ON_ERROR_KEY, "true");
+      CommitStep setPermissionCommitStep = new SetPermissionCommitStep(targetFs, ancestorOwnerAndPermissions, props);
```

Comment:
Why do we need this step? if dir already exist on dst and user want to change the permission for that dir, it will appear in the manifest? If dir not exist, we should directly create parent dir to be the same as source dir?

Reply:
If dir does not exist, we create it, but we create it with execute bit set. I do not want to not do that, because execute bit may be necessary to do publish.
So, in this PR, I had to add a post publish step to set the original permission.
Honestly, the requirement for this work, unsetting execute bit, seems quite odd.